src/wp-includes/class-wp-roles.php

```diff
@@ -160,6 +160,13 @@ public function add_role( $role, $display_name, $capabilities = array() ) {
 			return;
 		}
 
+		foreach( $capabilities as $key => $value ) {
+			if(!is_bool($value)) {
```

Comment:
Since this reuses `$value` as a key on lines 165 & 166, it would make the code more resilient to do: `is_int( $key ) && is_string( $value )`

But, I think if we are considering doing argument validation, it may warrant a dedicated method – one where we can more intentionally conclude (and document) what mutations will take place to what kind of submitted keys & values.

(The obvious benefit of a dedicated method would be reuse elsewhere that an array of caps is passed in as an argument, ensuring the same results, and ability to write unit tests for it.)

Alternatively, `wp_is_numeric_array()` could be used before the new `foreach`, but that function checks that *all* array keys are numeric, which means we’d lose support for mixed key types (which was an example in the core ticket).